src/scene/graphics/shared/svg/SVGToGraphicsPath.ts

```diff
@@ -1,244 +1,202 @@
+/* eslint-disable no-console */
+import parse from 'parse-svg-path';
 import { warn } from '../../../../utils/logging/warn';
 
 import type { GraphicsPath } from '../path/GraphicsPath';
 
-const commandSizeMap = {
-    a: 7,
-    c: 6,
-    h: 1,
-    l: 2,
-    m: 2,
-    q: 4,
-    s: 4,
-    t: 2,
-    v: 1,
-    z: 0,
-};
+interface SubPath
+{
+    startX: number;
+    startY: number;
+}
 
 // TODO optimise and cache the paths?
 export function SVGToGraphicsPath(svgPath: string, path: GraphicsPath): GraphicsPath
 {
-    // get commands removing spaces at the end
-
-    const commands = svgPath.match(/[a-df-z][^a-df-z]*/gi);
-    // get command data size
-
-    const data = svgPath.match(/[+-]?\d*\.?\d+(?:[eE][+-]?\d+)?/g)?.map(parseFloat);
-
-    const betterCommands: string[] = [];
+    const commands = parse(svgPath);
 
-    commands.forEach((command) =>
-    {
-        const data = command.match(/[+-]?\d*\.?\d+(?:[eE][+-]?\d+)?/g)?.map(parseFloat);
-
-        const type = command[0];
-
-        let totalInstructions = 1;
-
-        if (data)
-        {
-            totalInstructions = data.length / commandSizeMap[type.toLowerCase() as keyof typeof commandSizeMap];
-        }
-
-        for (let i = 0; i < totalInstructions; i++)
-        {
-            betterCommands.push(type);
-        }
-    });
-
-    let dataIndex = 0;
+    const subpaths: SubPath[] = [];
+    let currentSubPath: SubPath | null = null;
 
     let lastX = 0;
     let lastY = 0;
 
-    for (let i = 0; i < betterCommands.length; i++)
+    for (let i = 0; i < commands.length; i++)
     {
-        const type = betterCommands[i];
-
-        // const command = commands[i];
-
-        //        const type = betterCommand.type;
+        const command = commands[i];
+        const [type, ...data] = command;
```

Comment:
this destructions actually clones and creates a new Array which will affect performance. Especially if the commands are large.

Reply:
Changed to explicit variables